go/adbc/driver/snowflake/connection.go

```diff
@@ -557,13 +587,204 @@ func (c *cnxn) populateMetadata(ctx context.Context, depth adbc.ObjectDepth, cat
+			}
+		} else {
+			fullyQualifiedConstraintName = getFullyQualifiedConstraintName(data.dbName, data.schema, data.tblName, data.constraintName)
+			if _, exists := fullyQualifiedConstraintSchemaLookup[fullyQualifiedConstraintName]; !exists {
+				fullyQualifiedConstraintSchemaLookup[fullyQualifiedConstraintName] = internal.ConstraintSchema{
+					ConstraintName:        data.constraintName,
+					ConstraintType:        data.constraintType,
+					ConstraintColumnNames: []string{data.colName},
+				}
+			} else {
+				constraintInfo := fullyQualifiedConstraintSchemaLookup[fullyQualifiedConstraintName]
+				constraintInfo.ConstraintColumnNames = append(constraintInfo.ConstraintColumnNames, data.colName)
+				fullyQualifiedConstraintSchemaLookup[fullyQualifiedConstraintName] = constraintInfo
+			}
+		}
+	}
+
+	for key, constraintSchema := range fullyQualifiedConstraintSchemaLookup {
+		catalogSchemaTable := getCatalogSchemaTableFromFullyQualifiedConstraintName(key)
+		constraintLookup[catalogSchemaTable] = append(constraintLookup[catalogSchemaTable], constraintSchema)
+	}
+
+	return constraintLookup, nil
+}
+
+func (c *cnxn) getConstraintsData(ctx context.Context, depth adbc.ObjectDepth, metadataRecords []internal.Metadata) ([]TableConstraint, error) {
+	if depth == adbc.ObjectDepthCatalogs || depth == adbc.ObjectDepthDBSchemas {
+		return nil, nil
+	}
+	availableConstraintTypes := getAvailableConstraintTypes(metadataRecords)
+	availableFullyQualifiedConstraints := getAvailableFullyQualifiedConstraints(metadataRecords)
+
+	var uniqueConstraintsData []TableConstraint
+	var primaryKeyConstraintsData []TableConstraint
+	var foreignKeyConstraintsData []TableConstraint
+	var err error
+
+	if availableConstraintTypes != nil {
+		if _, exists := availableConstraintTypes[internal.Unique]; exists {
+			uniqueConstraintsData, err = c.getUniqueConstraints(ctx, availableFullyQualifiedConstraints)
+			if err != nil {
+				return nil, errToAdbcErr(adbc.StatusIO, err)
+			}
+		}
+
+		if _, exists := availableConstraintTypes[internal.PrimaryKey]; exists {
+			primaryKeyConstraintsData, err = c.getPrimaryKeyConstraints(ctx, availableFullyQualifiedConstraints)
+			if err != nil {
+				return nil, errToAdbcErr(adbc.StatusIO, err)
+			}
+		}
+
+		if _, exists := availableConstraintTypes[internal.ForeignKey]; exists {
+			foreignKeyConstraintsData, err = c.getForeignKeyConstraints(ctx, availableFullyQualifiedConstraints)
+			if err != nil {
+				return nil, errToAdbcErr(adbc.StatusIO, err)
+			}
+		}
+	}
+
+	tableConstraintsData := append(append(uniqueConstraintsData, primaryKeyConstraintsData...), foreignKeyConstraintsData...)
+
+	sort.Slice(tableConstraintsData, func(i, j int) bool {
+		if tableConstraintsData[i].constraintName == tableConstraintsData[j].constraintName {
+			return tableConstraintsData[i].keySequence < tableConstraintsData[j].keySequence
+		}
+		return tableConstraintsData[i].constraintName == tableConstraintsData[j].constraintName
+	})
```

Comment:
`slices.SortFunc` is likely faster and easier to use

Reply:
updated to use `slices.SortFunc`. Thanks for the suggestion!